Source/DafnyCore/DafnyOptions.cs

```diff
@@ -15,28 +19,87 @@
 using Bpl = Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
+  public enum FunctionSyntaxOptions {
+    Version3,
+    Migration3To4,
+    ExperimentalTreatUnspecifiedAsGhost,
+    ExperimentalTreatUnspecifiedAsCompiled,
+    ExperimentalPredicateAlwaysGhost,
+    Version4,
+  }
+
+  public enum QuantifierSyntaxOptions {
+    Version3,
+    Version4,
+  }
 
-  public record Options(IDictionary<IOptionSpec, object> OptionArguments);
+  public record Options(IDictionary<Option, object> OptionArguments);
 
   public class DafnyOptions : Bpl.CommandLineOptions {
+
+    public void ApplyBinding(Option option) {
+      if (bindings.ContainsKey(option)) {
+        bindings[option](this, Get(option));
+      }
+    }
+
+    public T Get<T>(Option<T> option) {
+      return (T)Options.OptionArguments.GetOrCreate(option, () => default(T));
+    }
+
+    public object Get(Option option) {
+      return Options.OptionArguments[option];
+    }
+
+    public void Set(Option option, object value) {
```

Comment:
Do we ever need this version given the more type-safe version below?

Reply:
It is used.